tests/integration/integration_test.py

```diff
@@ -939,6 +746,157 @@ def my_on_revoke(consumer, partitions):
+            'api.version.request': api_version_request,
+            'schema.registry.url': schema_registry_url
+            }
+
+    consumer_conf = dict({'group.id': generate_group_id(),
+                          'session.timeout.ms': 6000,
+                          'enable.auto.commit': False,
+                          'default.topic.config': {
+                              'auto.offset.reset': 'earliest'
+                          }}, **base_conf)
+
+    print('-' * 10, 'Verifying basic auth source USER_INFO', '-' * 10)
+    run_avro_loop(dict(base_conf, **user_info), dict(consumer_conf, **user_info))
+
+    print('-' * 10, 'Verifying basic auth source SASL_INHERIT', '-' * 10)
+    run_avro_loop(dict(base_conf, **sasl_inherit), dict(consumer_conf, **sasl_inherit))
+
+    print('-' * 10, 'Verifying basic auth source URL', '-' * 10)
+    run_avro_loop(dict(base_conf, **url), dict(consumer_conf, **url))
+
+
+def run_avro_loop(producer_conf, consumer_conf):
+    from confluent_kafka import avro
+    avsc_dir = os.path.join(os.path.dirname(__file__), os.pardir, 'avro')
+
+    p = avro.AvroProducer(producer_conf)
+
+    prim_float = avro.load(os.path.join(avsc_dir, "primitive_float.avsc"))
+    prim_string = avro.load(os.path.join(avsc_dir, "primitive_string.avsc"))
+    basic = avro.load(os.path.join(avsc_dir, "basic_schema.avsc"))
+    str_value = 'abc'
+    float_value = 32.0
+
+    combinations = [
+        dict(key=float_value, key_schema=prim_float),
+        dict(value=float_value, value_schema=prim_float),
+        dict(key={'name': 'abc'}, key_schema=basic),
+        dict(value={'name': 'abc'}, value_schema=basic),
+        dict(value={'name': 'abc'}, value_schema=basic, key=float_value, key_schema=prim_float),
+        dict(value={'name': 'abc'}, value_schema=basic, key=str_value, key_schema=prim_string),
+        dict(value=float_value, value_schema=prim_float, key={'name': 'abc'}, key_schema=basic),
+        dict(value=float_value, value_schema=prim_float, key=str_value, key_schema=prim_string),
+        dict(value=str_value, value_schema=prim_string, key={'name': 'abc'}, key_schema=basic),
+        dict(value=str_value, value_schema=prim_string, key=float_value, key_schema=prim_float),
+        # Verify identity check allows Falsy object values(e.g., 0, empty string) to be handled properly (issue #342)
+        dict(value='', value_schema=prim_string, key=0.0, key_schema=prim_float),
+        dict(value=0.0, value_schema=prim_float, key='', key_schema=prim_string),
+    ]
+
+    for i, combo in enumerate(combinations):
+        combo['topic'] = str(uuid.uuid4())
+        combo['headers'] = [('index', str(i))]
+        p.produce(**combo)
+    p.flush()
+
+    c = avro.AvroConsumer(consumer_conf)
+    c.subscribe([(t['topic']) for t in combinations])
+
+    msgcount = 0
+    while msgcount < len(combinations):
+        msg = c.poll(100)
+
+        if msg is None or msg.error():
+            continue
```

Comment:
I believe it is valuable, for troubleshooting purposes, to print errors, even if we ignore them